### ISSUE TITLE
Confusing 3.0 error message: Skipping undefined keyspace: view_pending_updates

### ISSUE THREAD

**Issue by veramine** [scylladb/scylla#4014]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- [X] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash):  3.0.rc3-0.20181223.dfb23f4

I was trying to track down a separate issue today and saw this error every time scylla 3.0 starts:

```
Dec 23 06:14:22 ip-172-31-39-55.us-west-2.compute.internal scylla[1947]:  [shard 0] database - Skipping undefined keyspace: view_pending_updates
```

I incorrectly inferred that view_pending_updates was a missing keyspace that might need to be created.  Avi told me it is just a hints directory, and suggested that I could file a bug suggesting that the message be made more clear in the future.  This is low priority though.  Thanks.

**Comment by nyh** [comment]
@slivne I sent a patch for this. Please add the backport-to-3.0 label on this so we'll also backport it to 3.0. As usual, I don't have the permissions :-(

**Comment by slivne** [comment]
@tzach lets release note this for 3.0.0

**Comment by nyh** [comment]
Update on the status of this issue:
I already sent a patch fixing this issue by avoiding treating a data subdirectory by the name `view_pending_updates` as a keyspace.
However, @avikivity didn't like this direction - he thought the underlying problem is more serious than just spurious warning messages - it will cause real problems if somebody accidentally decides to call a keyspace by the specific name `view_pending_updates`. So he asked that instead, we should move this directory from the "data" directory into the "commitlog" directory. I can easily do this, but was always preempted by more urgent stuff, so let's just decide we want to fix this Avi's way, and I will.
CC @eliransin @slivne @duarten @psarna

**Comment by nyh** [comment]
Oh, and I think it's a mistake to postpone moving the view_pending_updates directory beyond 3.0.0. If we do, we'll later need to support upgrades and both locations...

**Comment by avikivity** [comment]
It's not just the name collision, scripts may expect directories in the data directory to behave like keyspaces, and break when they don't.

**Comment by nyh** [comment]
Fixed by commit da090a5458a436935d7e9a29d13f2d896e27eefa
Sorry for forgetting the "Fixes" string in the commit message. Please close this issue.
